src/main.rs

```diff
@@ -272,7 +272,20 @@ async fn run_build_from_args(args: BuildOpts) -> anyhow::Result<()> {
             tracing::error!("Could not flatten selectors");
         }
 
-        let recipe = render_recipe(&recipe_yaml, &variant, &hash)?;
+        let recipe = match render_recipe(&recipe_yaml, &variant, &hash) {
+            Result::Err(e) => {
+                match &e {
+                    render::recipe::RecipeRenderError::InvalidYaml(inner) => {
+                        tracing::error!("Failed to parse recipe YAML: {}", inner.to_string());
+                    }
+                    render::recipe::RecipeRenderError::YamlNotMapping => {
+                        tracing::error!("{}", e);
+                    }
+                }
+                return Err(e.into());
```

Comment:
returning it again as an error prints it twice in the terminal. The tracing one is nice as it prints in red plus the source crate is mentioned.